tests/helper/helper_generic.go

```diff
@@ -312,12 +317,45 @@ func CommonBeforeEach() CommonVar {
 
 // CommonAfterEach is common function that cleans up after every test Spec (It)
 func CommonAfterEach(commonVar CommonVar) {
+	// Get details, including test result for each test spec and adds it to local testResults.txt file
+	// Ginkgo test related variables
+	commonVar.testFileName = strings.Replace(CurrentGinkgoTestDescription().FileName[strings.LastIndex(CurrentGinkgoTestDescription().FileName, "/")+1:strings.LastIndex(CurrentGinkgoTestDescription().FileName, ".")], "_", "-", -1) + ".go"
+	commonVar.testCase = CurrentGinkgoTestDescription().FullTestText
+	commonVar.testFailed = CurrentGinkgoTestDescription().Failed
+	commonVar.testDuration = CurrentGinkgoTestDescription().Duration.Seconds()
+
+	var prNum string
+	var resultsRow string
+	prNum = os.Getenv("GIT_PR_NUMBER")
+	passedOrFailed := "PASSED"
+	if commonVar.testFailed {
+		passedOrFailed = "FAILED"
+	}
+	clusterType := "OCP"
+	if IsKubernetesCluster() {
+		clusterType = "KUBERNETES"
+	}
+	testDate := strings.Split(time.Now().Format(time.RFC3339), "T")[0]
+	resultsRow = prNum + "," + testDate + "," + clusterType + "," + commonVar.testFileName + "," + commonVar.testCase + "," + passedOrFailed + "," + strconv.FormatFloat(commonVar.testDuration, 'E', -1, 64) + "\n"
+	testResultsFile := filepath.Join("/", "tmp", "testResults.txt")
+
+	f, err := os.OpenFile(testResultsFile, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		fmt.Println("Error: ", err)
+	} else {
+		defer f.Close()
+		if _, err = f.WriteString(resultsRow); err != nil {
+			fmt.Println("Error: ", err)
+		}
+		f.Close()
+	}
+
```

Comment:
**Optional:**
```suggestion
                _, err = f.WriteString(resultsRow) 
 		if err != nil {
 			fmt.Println("Error: ", err)
 		}

		if err = f.Close(); err != nil {
 			fmt.Println("Error: ", err)
 		}
```
Since we are performing write operation, it is advised that it is done in a separate line, because it makes some changes, for a normal read operation, an inline read and error check is fine.

Reply:
Suggestion applied